Description: Fix symbol clash when including cl2.hpp from multiple source files
Author: James Price <j.price@bristol.ac.uk>
Last-Update: 2015-10-24
--- a/cl2.hpp
+++ b/cl2.hpp
@@ -6557,7 +6557,7 @@
     OutOfOrder = CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
 };
 
-QueueProperties operator|(QueueProperties lhs, QueueProperties rhs)
+inline QueueProperties operator|(QueueProperties lhs, QueueProperties rhs)
 {
     return static_cast<QueueProperties>(static_cast<cl_command_queue_properties>(lhs) | static_cast<cl_command_queue_properties>(rhs));
 }
@@ -8080,7 +8080,7 @@
     Profiling = CL_QUEUE_PROFILING_ENABLE,
 };
 
-DeviceQueueProperties  operator|(DeviceQueueProperties  lhs, DeviceQueueProperties  rhs)
+inline DeviceQueueProperties operator|(DeviceQueueProperties lhs, DeviceQueueProperties rhs)
 {
     return static_cast<DeviceQueueProperties>(static_cast<cl_command_queue_properties>(lhs) | static_cast<cl_command_queue_properties>(rhs));
 }
